dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/DataExtension.java

```diff
@@ -454,13 +479,11 @@ private static Class<?> requirePrimaryEntity(Class<?> primaryEntityClass,
                                                  Class<?> repositoryInterface,
                                                  Method method) {
         if (primaryEntityClass == null)
-            throw new MappingException("An entity class is needed by the " + method.getName() +
-                                       " method, but the " + repositoryInterface.getName() +
-                                       " repository interface does not specify a primary entity class." +
-                                       " To correct this, have the " + repositoryInterface.getSimpleName() +
-                                       " interface extend one of the built-in repository supertypes," +
-                                       " such as DataRepository<EntityClass, KeyClass>, supplying the" +
-                                       " entity class as the first type parameter."); // TODO NLS
+            throw exc(MappingException.class,
+                      "CWWKD1001.no.primary.entity",
+                      method.getName(),
+                      repositoryInterface.getName(),
+                      "DataRepository<EntityClass, KeyClass>");
```

Comment:
```suggestion
                      "DataRepository<EntityClass, IdClass>");
```

In most all cases we avoid using `key` and instead refer to `id` in Jakarta Data

Reply:
I didn't use IdClass here because it happens to be a Jakarta Persistence class name (for a particular type of composite id) and I thought that would be confusing.